src/libmeasurement_kit/net/libssl.hpp

```diff
@@ -288,42 +288,36 @@ template <size_t max_cache_size = 64> class Cache {
 };
 
 /*!
-    \brief Verify SSL peer.
+    \brief Set hostname for verifying the peer certificate.
 
-    \param hostname Expected SNI hostname.
+    \param hostname Expected hostname.
 
     \param ssl Pointer to SSL struct.
 
     \param logger Logger used for logging.
 
     \return NoError() on success, an error on failure.
 */
-template <MK_MOCK(SSL_get_verify_result), MK_MOCK(SSL_get_peer_certificate),
-        MK_MOCK(tls_check_name)>
-Error verify_peer(std::string hostname, SSL *ssl, SharedPtr<Logger> logger) {
-    assert(ssl != nullptr);
-    logger->debug("SSL version: %s", SSL_get_version(ssl));
-    long verify_err = SSL_get_verify_result(ssl);
-    if (verify_err != X509_V_OK) {
-        logger->warn("ssl: got an invalid certificate");
-        return SslInvalidCertificateError(
-                X509_verify_cert_error_string(verify_err));
-    }
-    X509 *server_cert = SSL_get_peer_certificate(ssl);
-    if (server_cert == nullptr) {
-        logger->warn("ssl: got no certificate");
-        return SslNoCertificateError();
+template <MK_MOCK(SSL_get0_param), MK_MOCK(X509_VERIFY_PARAM_set1_host)>
+Error set_hostname_for_verification(
```

Comment:
I believe the name of this function is a bit misleading. Plus, the code to really enable peer verification should probably be refactored inside here, for additional clarity. Plus a link to the OpenSSL Wiki is required.